packages/components/modules/payments/web/hooks/useStripeHook.tsx

```diff
@@ -1,50 +1,94 @@
 import { useNotification } from '@baseapp-frontend/utils'
 
 import { Stripe } from '@stripe/stripe-js'
-import { useMutation, useQuery } from '@tanstack/react-query'
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
 
-import {
-  CONFIRM_CARD_PAYMENT_API_KEY,
-  PAYMENT_METHOD_API_KEY,
-  PRODUCT_API_KEY,
-} from '../services/keys'
-import PaymentMethodApi from '../services/stripe'
+import { CONFIRM_CARD_PAYMENT_API_KEY } from '../services/keys'
+import StripeApi from '../services/stripe'
 import { CreateSubscriptionOptions } from '../types'
 
 const useStripeHook = () => {
   const { sendToast } = useNotification()
+  const queryClient = useQueryClient()
+
+  const useGetCustomer = () =>
+    useQuery({
+      queryKey: ['useGetCustomer'],
+      queryFn: () => StripeApi.getCustomer(),
+    })
+
+  const useCreateCustomer = () =>
+    useMutation({
+      mutationFn: (userId?: string) => StripeApi.createCustomer(userId),
+    })
 
   const useSetupIntent = (customerId?: string) =>
     useMutation({
-      mutationFn: (id: string = customerId || '') => PaymentMethodApi.createSetupIntent(id),
+      mutationFn: (id: string = customerId || '') => StripeApi.createSetupIntent(id),
+      onSuccess: () => {
+        console.log('Setup intent created successfully:')
+      },
       onError: (error) => {
         sendToast(error.message, { type: 'error' })
       },
       mutationKey: ['useSetupIntent', customerId],
     })
 
-  const useGetPaymentMethod = (customerId: string) =>
+  const useListPaymentMethods = (customerId: string) =>
     useQuery({
-      queryKey: [PAYMENT_METHOD_API_KEY.get(), customerId],
-      queryFn: () => PaymentMethodApi.getPaymentMethod(customerId),
+      queryKey: ['listPaymentMethods', customerId],
```

Comment:
Let's move this queryKey to the service/keys file and apply it as a function, the same way we did for CONFIRM_CARD_PAYMENT_API_KEY:

export const CONFIRM_CARD_PAYMENT_API_KEY = {
default: ['confirmCardPayment'] as QueryKey,
get: (...params: string[]) => [...CONFIRM_CARD_PAYMENT_API_KEY.default, ...params] as QueryKey,
}